spec/requests/posts_controller_spec.rb

```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+RSpec.describe 'PostsController', type: :request do
+  let(:user) { create(:user) } # Create user from factory
+
+  it 'renders the index action' do
+    get user_posts_path(user_id: user.id)
+    expect(response).to have_http_status(:ok)
+    expect(response).to render_template(:index)
+  end
+
+  it 'renders the show action' do
+    post = create(:post, user:) # Create post from factory
+    get user_post_path(user_id: user.id, id: post.id)
+    expect(response).to have_http_status(:ok)
+    expect(response).to render_template(:show)
+  end
```

Comment:
- [ ] As I mentioned in the above comment and for the project requirement, kindly ensure that each action file include tests that for each action you include test for the correct placeholder text of the body
Example : expect(response.body).to include("Welcome to user index page")

- [ **OPTIONAL** ] For best practices and clean code, I think it would be better if all tests related to an `action` to be include in a `describe` method and each specific test in `it` method. And ensure you apply this approach in all files(i.e. `spec/requests/posts_controller_spec.rb` and `spec/requests/users_controller_spec.rb`). For example:
```ruby
describe 'GET /index' do
   it 'should render the correct template' do
    expect(response).to have_http_status(:ok)
  end
  
  it 'should show the correct placeholder tex' do
    ...
  end
  .
  .
  .
end

describe 'GET /show' do
  ...
end
```